internal/lru/lru.go

```diff
@@ -55,40 +61,69 @@ func New(maxEntries int) *Cache {
 }
 
 // Add adds a value to the cache.
-func (c *Cache) Add(key string, value interface{}) {
-	if c.cache == nil {
-		c.cache = make(map[string]*list.Element)
-		c.ll = list.New()
-	}
-	if ee, ok := c.cache[key]; ok {
-		c.ll.MoveToFront(ee)
-		ee.Value.(*entry).value = value
+func (c *Cache) Add(key string, val interface{}) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	if ele, ok := c.cache[key]; ok {
+		c.ll.MoveToFront(ele)
+		ele.Value.(*entry).value = val
 		return
 	}
-	ele := c.ll.PushFront(&entry{key, value})
-	c.cache[key] = ele
-	if c.MaxEntries != 0 && c.ll.Len() > c.MaxEntries {
-		c.RemoveOldest()
-	}
+	c.add(key, val)
 }
 
 // Get looks up a key's value from the cache.
 func (c *Cache) Get(key string) (value interface{}, ok bool) {
-	if c.cache == nil {
-		return
+	timestamp := time.Now()
+	c.lock.RLock()
+	if ele, ok := c.cache[key]; ok {
+		ent := ele.Value.(*entry)
+		if timestamp.After(ent.timestamp.Add(time.Second * 1)) {
```

Comment:
We should probably make this magic number configurable via session or something.

Reply:
We can inject it via config vars.